File: tests/fixtures/codegen/inputs/standalone-expected/typescript/runtime/queries/items/ReturningPayload.ts
import type { ReturningPayloadParams, ReturningPayloadRow } from "../../../types/queries/items/ReturningPayload.js";
export type { ReturningPayloadParams, ReturningPayloadRow } from "../../../types/queries/items/ReturningPayload.js";
import * as pgsid from "./pgsid/queryable.js";
export { QueryValidationError } from "./pgsid/queryable.js";
import * as _jsonSchemas from "../../jsonschemas/index.js";
export const returningPayloadSql = "UPDATE items SET payload = $1 RETURNING payload;";
export function isReturningPayloadPayload(value: unknown): value is ReturningPayloadRow["payload"] {
    return _jsonSchemas.isEvent(value);
}
export function validateReturningPayloadPayload(value: unknown): {
    valid: boolean;
    issues: {
        path: readonly (string | number)[];
        keyword: string;
        expected: unknown;
        received: string;
        message: string;
    }[];
} {
    return _jsonSchemas.validateEvent(value);
}
export function isReturningPayloadParamsPayloadPublicItemsPayload(value: unknown): value is ReturningPayloadParams["payload"] {
    return _jsonSchemas.isEvent(value);
}
export function validateReturningPayloadParamsPayloadPublicItemsPayload(value: unknown): {
    valid: boolean;
    issues: {
        path: readonly (string | number)[];
        keyword: string;
        expected: unknown;
        received: string;
        message: string;
    }[];
} {
    return _jsonSchemas.validateEvent(value);
}
export async function returningPayload(db: pgsid.Queryable, params: ReturningPayloadParams): Promise<ReturningPayloadRow | undefined> {
    const jsonInput1 = JSON.stringify(params["payload"]);
    if (jsonInput1 === undefined)
        throw new pgsid.QueryValidationError("ReturningPayload", "payload", [{ path: [], keyword: "json", expected: "JSON value", received: "undefined", message: "Value has no JSON representation" }]);
    const jsonInput1Value = jsonInput1 === null ? null : JSON.parse(jsonInput1);
    if (jsonInput1 !== null) {
        if (!isReturningPayloadParamsPayloadPublicItemsPayload(jsonInput1Value))
            throw new pgsid.QueryValidationError("ReturningPayload", "payload", validateReturningPayloadParamsPayloadPublicItemsPayload(jsonInput1Value).issues);
    }
    const result = await db.query(returningPayloadSql, [jsonInput1]);
    const row = result.rows[0];
    if (row === undefined)
        return undefined;
    if (!isReturningPayloadPayload(row["payload"]))
        throw new pgsid.QueryValidationError("ReturningPayload", "payload", validateReturningPayloadPayload(row["payload"]).issues);
    return row as ReturningPayloadRow;
}

File: tests/fixtures/codegen/inputs/structs-expected/queries/items/returningpayload.go
package items

import (
	"context"
	jsonschemas "example.com/pgsid-validation/generated/jsonschemas"
	pgsidpgx "example.com/pgsid-validation/generated/queries/pgsid/pgx"
)

const ReturningPayloadSQL = "UPDATE items SET payload = $1 RETURNING payload;"

type ReturningPayloadParams struct {
	Payload jsonschemas.Event `db:"payload"`
}
type ReturningPayloadRow struct {
	Payload jsonschemas.Event `db:"payload"`
}

func (q *Queries) ReturningPayload(ctx context.Context, params ReturningPayloadParams) (ReturningPayloadRow, error) {
	jsonInput1, err := pgsidpgx.ValidateJSONInput(params.Payload, "pgsid:///jsonschemas/Event.json#", "ReturningPayload", "payload", false)
	if err != nil {
		return ReturningPayloadRow{}, err
	}
	var row ReturningPayloadRow
	err = q.db.QueryRow(ctx, ReturningPayloadSQL, jsonInput1).Scan(pgsidpgx.ScanTargets(pgsidpgx.ValidatedJSON(&row.Payload, "pgsid:///jsonschemas/Event.json#", "ReturningPayload", "payload", false)))
	return row, err
}
